raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -409,6 +446,22 @@ private EndQuorumEpochResponseData handleEndQuorumEpochRequest(EndQuorumEpochReq
         return buildEndQuorumEpochResponse(Errors.NONE);
     }
 
+
+    private Optional<Errors> handleEndQuorumEpochResponse(
+        RaftResponse.Inbound responseMetadata
+    ) throws IOException {
+        EndQuorumEpochResponseData response = (EndQuorumEpochResponseData) responseMetadata.data;
+        Errors error = Errors.forCode(response.errorCode());
+
+        if (handleNonMatchingResponseLeaderAndEpoch(response.leaderEpoch(), optionalLeaderId(response.leaderId()))) {
+            return Optional.of(error);
+        } else if (error == Errors.NONE) {
```

Comment:
For both Begin / EndQuorum request, we can also return FENCED_LEADER_EPOCH and INVALID_REQUEST which should be handled before `handleUnexpectedError` right? I think we should not just backoff on connection cache for these two?